ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -150,15 +154,15 @@ Tensor create_tt_tensor_from_py_data(
     std::size_t py_data_ptr,
     const TensorSpec& tensor_spec,
     Device* device,
-    bool override_enable_borrow,
+    const bool override_enable_borrow,
     const std::function<void()>& on_creation_callback,
     const std::function<void()>& on_destruction_callback) {
     auto layout = tensor_spec.layout();
 
-    bool enable_borrow = true;
-    if (layout != Layout::ROW_MAJOR or override_enable_borrow) {
-        enable_borrow = false;
-    }
+    const bool requires_padding = tensor_spec.logical_shape().volume() !=
+                                  tensor_spec.physical_shape().height() * tensor_spec.physical_shape().width();
+    const bool requires_tilization = layout != Layout::ROW_MAJOR;
+    const bool enable_borrow = !(requires_padding or requires_tilization or override_enable_borrow);
```

Comment:
fyi @omilyutin-tt since you're touching this code

Reply:
Yeah, @TT-BrianLiu I just renamed `override_enable_borrow` to `force_disable_borrow` in a different PR... This change makes sense, I'd just rewrite the `const bool enable_borrow = !requires_padding and !requires_tilization and !override_enable_borrow;` seems more natural.

Reply:
sounds good, I can update. Is it safe to assume my PR going in first?

Reply:
Mine is already in:) It just had that one rename though, if you do the same rename and just override with your version when resolving conflicts, that should be it.